addon/serializers/json-api.js

```diff
@@ -141,8 +141,20 @@ const JSONAPISerializer = JSONSerializer.extend({
     @private
   */
   _normalizeRelationshipDataHelper(relationshipDataHash) {
-    let type = this.modelNameFromPayloadKey(relationshipDataHash.type);
-    relationshipDataHash.type = type;
+    let modelName = this.modelNameFromPayloadType(relationshipDataHash.type);
+    let deprecatedModelNameLookup = this.modelNameFromPayloadKey(relationshipDataHash.type);
+
+    if (modelName !== deprecatedModelNameLookup && this.modelNameFromPayloadType === JSONAPISerializer.prototype.modelNameFromPayloadType) {
```

Comment:
It seems like if a user intentionally wanted `modelNameFromPayloadType` and `modelNameFromPayloadKey` to return different values and they needed to define a custom `modelNameFromPayloadType` we would always hit the deprecated case.

What about if we always called `modelNameFromPayloadType` but had its default implementation just call `modelNameFromPayloadKey`?